NEWS.md

```diff
@@ -1,5 +1,9 @@
 # Release Notes
 
+The current stable version of Nicotine+ is 3.2.9, released on March 5, 2023, see [below](#version-329-march-5-2023).
```

Comment:
I'm not a fan of adding another place where a date has to be updated. I'd keep the "The current stable version of Nicotine+ is 3.2.9" part, and add the link to DOWNLOADS.md after.